components/operator-service/src/main/java/com/wso2telco/dep/operatorservice/dao/BlackListWhiteListDAO.java

```diff
@@ -178,35 +196,210 @@ public void blacklist(MSISDNValidationDTO msisdns, final String apiID, final Str
 		return returnList;
 	}
 
-	public String[] getBlacklisted(MSISDNSearchDTO searchDTO) throws Exception {
+	/**
+	 * Retrieves the number of blacklist entries for a particular API ID
+	 * @return
+	 * @throws Exception
+	 */
+	public int getBlacklistCountByApi(BlacklistWhitelistDTO dto) throws Exception {
+		StringBuilder sql = new StringBuilder();
 
+		sql.append("SELECT count(*) as count");
```

Comment:
No point of dividing in to, very small parts. Since we are using this in a wrong way. Better to define all the sqls in one location and access sql from it. 
the proper way is, move all the sqls in to a property file and access from it.